lib/utilities/addon-calculate-cache-key-for-tree.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+var crypto = require('crypto');
+
+module.exports = function calculateCacheKeyForTree(treeType, addonInstance) {
+  var cacheKeyParts = [
+    addonInstance.pkg,
```

Comment:
It is much simpler to simply checksum the full `package.json` here for sure.  It costs us little, and reduces the complexity of this quite a bit.

One thing to note (I added an inline comment for this):

>  In a future refactor, this function will be extracted and migrated to a
>  shared package. This allows addon authors to easily utilize the "default"
>  cache key generation system (e.g. if they implement `treeFor*` in a way that
>  is totally safe), but still allows us to identify issues with the cache key
>  generation and iterate (since both addon and ember-cli could use `^1.0.0`
>  and float independently).